src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,139 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private MyNode<T> first;
+    private MyNode<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            MyNode<T> newNode = new MyNode<>(null, value, null);
+            first = newNode;
+            last = newNode;
+        } else {
+            MyNode<T> l = last;
+            MyNode<T> newNode = new MyNode<>(l, value, null);
+            last = newNode;
+            l.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        MyNode<T> oldNode = findNodeByIndex(index);
+        if (oldNode == first) {
+            MyNode<T> f = first;
+            MyNode<T> newNode = new MyNode<>(null, value, f);
+            first = newNode;
+            f.prev = newNode;
+        } else {
+            MyNode<T> l = oldNode.prev;
+            MyNode<T> newNode = new MyNode<>(l, value, oldNode);
+            oldNode.prev = newNode;
+            l.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
+            add(node);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        MyNode<T> resultNode = findNodeByIndex(index);
+        return resultNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        MyNode<T> current = findNodeByIndex(index);
+        T result = current.item;
+        current.item = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        MyNode<T> nodeToRemove = findNodeByIndex(index);
+        unlinkNode(nodeToRemove);
+        return nodeToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        MyNode<T> f = first;
+        while (f != null) {
+            if (Objects.equals(f.item, object)) {
```

Comment:
Please, do not use class Objects
see [common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-use-class-objects)